internal/pkg/manifest/svc.go

```diff
@@ -550,76 +550,49 @@ func sortExposedPorts(exposedPorts []ExposedPort) []ExposedPort {
-	exposedPorts, err := s.ExposedPorts()
-	if err != nil {
-		return "", "", err
-	}
+func (rr *RoutingRule) Target(exposedPorts ExposedPortsIndex) (targetContainer string, targetPort string, err error) {
 	// Route load balancer traffic to main container by default.
-	targetContainer = aws.StringValue(s.Name)
-	targetPort = s.MainContainerPort()
-
-	rrTargetContainer := s.RoutingRule.Main.TargetContainer
-	rrTargetPort := s.RoutingRule.Main.TargetPort
-	if rrTargetContainer == nil && rrTargetPort == nil { // both targetPort and targetContainer are nil.
-		return
-	}
-
-	if rrTargetPort == nil { // when target_port is nil
-		if rrTargetContainer != s.Name {
-			targetContainer = aws.StringValue(rrTargetContainer)
-			targetPort = aws.StringValue(s.Sidecars[aws.StringValue(rrTargetContainer)].Port)
-		}
-		return
-	}
-
-	if rrTargetContainer == nil { // when target_container is nil
-		container, port := targetContainerFromTargetPort(exposedPorts, rrTargetPort)
-		targetPort = aws.StringValue(port)
-		if container != nil {
-			targetContainer = aws.StringValue(container)
+	targetContainer = exposedPorts.WorkloadName
+	for _, portConfig := range exposedPorts.PortsForContainer[targetContainer] {
+		if portConfig.isDefinedByContainer {
+			targetPort = strconv.Itoa(int(portConfig.Port))
 		}
-		return
-	}
-
-	// when both target_port and target_container are not nil
-	targetContainer = aws.StringValue(rrTargetContainer)
-	targetPort = template.StrconvUint16(aws.Uint16Value(rrTargetPort))
-
-	return
-}
-
-// HTTPLoadBalancerTarget returns target container and target port for the ALB configuration.
-// This method should be called only when ALB config is not empty.
-func (s *BackendService) HTTPLoadBalancerTarget() (targetContainer string, targetPort string, err error) {
-	exposedPorts, err := s.ExposedPorts()
-	if err != nil {
-		return "", "", err
 	}
 
-	// Route load balancer traffic to main container by default.
-	targetContainer = aws.StringValue(s.Name)
-	targetPort = s.MainContainerPort()
-
-	rrTargetContainer := s.RoutingRule.Main.TargetContainer
-	rrTargetPort := s.RoutingRule.Main.TargetPort
+	rrTargetContainer := rr.TargetContainer
+	rrTargetPort := rr.TargetPort
 	if rrTargetContainer == nil && rrTargetPort == nil { // both targetPort and targetContainer are nil.
 		return
 	}
 
 	if rrTargetPort == nil { // when target_port is nil
-		if rrTargetContainer != s.Name {
+		if aws.StringValue(rrTargetContainer) != exposedPorts.WorkloadName {
 			targetContainer = aws.StringValue(rrTargetContainer)
-			targetPort = aws.StringValue(s.Sidecars[aws.StringValue(rrTargetContainer)].Port)
+			for _, portConfig := range exposedPorts.PortsForContainer[targetContainer] {
+				if portConfig.isDefinedByContainer {
+					targetPort = strconv.Itoa(int(portConfig.Port))
+					/* NOTE: When the `target_port` is empty, the intended target port should be the port that is explicitly exposed by the container. Consider the following example
```

Comment:
Sorry, sorry, more comments. Can we make sure to document this super clearly on the site?